pyvista/core/pointset.py

```diff
@@ -2497,6 +2497,31 @@ def dimensions(self, dims):  # numpydoc ignore=GL08
         self.SetDimensions(nx, ny, nz)
         self.Modified()
 
+    @property
+    def dimensionality(self) -> int:
+        """Return the dimensionality of the grid.
+
+        Returns
+        -------
+        int
+            The grid dimensionality.
+
+        Examples
+        --------
+        >>> import pyvista as pv
+        >>> import numpy as np
+        >>> xrng = np.arange(-10, 10, 1, dtype=np.float32)
+        >>> yrng = np.arange(-10, 10, 2, dtype=np.float32)
+        >>> zrng = np.arange(-10, 10, 5, dtype=np.float32)
+        >>> x, y, z = np.meshgrid(xrng, yrng, zrng, indexing='ij')
+        >>> grid = pv.StructuredGrid(x, y, z)
+        >>> grid.dimensionality
+        3
+
+        """
+        dims = np.asarray(self.dimensions)
```

Comment:
Codecov produces a missing coverage but I don´t  know why since the lines are covered, see https://app.codecov.io/gh/pyvista/pyvista/pull/6676

Reply:
I checked the codecov link, and it seems that the tests are covered. Is it okay to merge this?

Reply:
For me, it's okay to merge it, since:
- I believe I have addressed all @user27182 comments
- I have been using the feature introduced by this PR for 2 weeks without issues

But ofc I have a conflict of interest there ;)

Reply:
We have waited long enough for this and will merge. Please open a new PR if someone wish to add something.